### ISSUE TITLE
Redesign MorphemeAwareEmbedding architecture for new training plan

### ISSUE THREAD

**Issue by marctjones** [marctjones/klareco#62]
## Parent Issue
Part of #58 (Redesigned Training Plan)

## Current State
`klareco/embeddings/morpheme_aware.py` has the right structure but needs updates to support:
1. Frozen grammatical features (not learned)
2. Separate training phases (roots first, then affixes)
3. Clear separation between semantic (learned) and grammatical (frozen) components

### 1. Add Frozen Grammatical Features
Replace learned embeddings with `register_buffer` for deterministic features:

```python
class GrammaticalFeatures(nn.Module):
    """Deterministic (frozen) features for grammatical morphemes."""
    
    def __init__(self, dim=16):
        super().__init__()
        # NOT nn.Embedding - these are FIXED
        self.register_buffer('ending_features', self._create_ending_features(dim))
        self.register_buffer('aspect_suffix_features', self._create_aspect_features(dim))
        self.register_buffer('gram_prefix_features', self._create_gram_prefix_features(dim))
    
    def _create_ending_features(self, dim):
        # Hand-crafted linguistic features, NOT random
        # Format: [word_class(4), case(2), number(2), tense(4), mood(2), voice(2)]
        features = {
            'o':   [1,0,0,0, 0,0, 0,0, 0,0,0,0, 0,0, 0,0],  # noun, nom, sing
            'oj':  [1,0,0,0, 0,0, 1,0, 0,0,0,0, 0,0, 0,0],  # noun, nom, plur
            'on':  [1,0,0,0, 1,0, 0,0, 0,0,0,0, 0,0, 0,0],  # noun, acc, sing
            'ojn': [1,0,0,0, 1,0, 1,0, 0,0,0,0, 0,0, 0,0],  # noun, acc, plur
            'a':   [0,1,0,0, 0,0, 0,0, 0,0,0,0, 0,0, 0,0],  # adj, nom, sing
            # ... etc
            'as':  [0,0,0,1, 0,0, 0,0, 1,0,0,0, 1,0, 0,0],  # verb, present, indicative
            'is':  [0,0,0,1, 0,0, 0,0, 0,1,0,0, 1,0, 0,0],  # verb, past, indicative
            'os':  [0,0,0,1, 0,0, 0,0, 0,0,1,0, 1,0, 0,0],  # verb, future, indicative
            # ... etc
        }
        return torch.tensor([features[e] for e in ENDING_ORDER], dtype=torch.float)
```

### 2. Update affix_classification.json
Add linguistic feature definitions:

```json
{
  "grammatical_features": {
    "endings": {
      "o":   {"word_class": "noun", "case": "nom", "number": "sing"},
      "as":  {"word_class": "verb", "tense": "present", "mood": "indicative"},
      ...
    },
    "aspect_suffixes": {
      "ad":  {"aspect": "continuative"},
      "ig":  {"voice": "causative"},
      "igh": {"voice": "inchoative"},
      ...
    }
  }
}
```

### 3. Support Phased Training
Add methods to freeze/unfreeze components:

```python
def freeze_roots(self):
    """Freeze root embeddings after Phase 1."""
    for param in self.root_embed.parameters():
        param.requires_grad = False

def freeze_grammatical(self):
    """Grammatical features are always frozen (buffers, not params)."""
    pass  # Already frozen by design

def get_trainable_params(self, phase: str):
    """Get parameters for specific training phase."""
    if phase == "roots":
        return self.root_embed.parameters()
    elif phase == "affixes":
        return chain(
            self.semantic_prefix_embed.parameters(),
            self.semantic_suffix_embed.parameters()
        )
```

## Tasks
- [ ] Add `GrammaticalFeatures` class with frozen buffers
- [ ] Update `affix_classification.json` with linguistic feature definitions  
- [ ] Add `freeze_roots()` and phase-aware parameter access
- [ ] Update `__init__` to use buffers for grammatical components
- [ ] Add unit tests for frozen behavior

**Comment by marctjones** [comment]
## Key Architecture Change: Frozen vs Learned

The redesigned architecture must use `register_buffer()` for grammatical features instead of `nn.Embedding()`.

### Why Buffers Instead of Embeddings?

```python
class MorphemeAwareEmbedding(nn.Module):
    def __init__(self):
        super().__init__()

        # TRAINED: Semantic components
        self.root_embeddings = nn.Embedding(5000, 128)
        self.semantic_prefix_embeddings = nn.Embedding(8, 128)
        self.semantic_suffix_embeddings = nn.Embedding(15, 128)

        # FROZEN: Grammatical features (never trained)
        ending_features = self._build_ending_features()
        self.register_buffer('ending_features', ending_features)
```

### Benefits of `register_buffer()`

1. **Saves with model** - included in state_dict
2. **Excluded from training** - no gradients computed
3. **Deterministic** - grammatical features are 100% known
4. **Efficient** - reduces trainable parameters

### Grammatical Feature Encoding

16-bit feature vectors encoding linguistic properties:

```python
def _build_ending_features(self):
    features = {
        'o':  [1,0,0,0, 0,0, 0,0, 0,0,0,0, 0,0,0,0],  # noun, nom, sg
        'on': [1,0,0,0, 0,1, 0,0, 0,0,0,0, 0,0,0,0],  # noun, acc, sg
        'oj': [1,0,0,0, 0,0, 0,1, 0,0,0,0, 0,0,0,0],  # noun, nom, pl
        'as': [0,0,0,1, 0,0, 0,0, 1,0,0,0, 0,0,0,0],  # verb, present
        'is': [0,0,0,1, 0,0, 0,0, 0,1,0,0, 0,0,0,0],  # verb, past
        # ... all 17 endings
    }
    return torch.tensor([features[e] for e in sorted(features)])
```

This is NOT random - each bit encodes a specific grammatical property (POS, case, number, tense).

### Component Classification Table

| Component | Count | Dim | Type | Training |
|-----------|-------|-----|------|----------|
| Roots | 5,000 | 128d | nn.Embedding | Phase 1 |
| Semantic Prefixes | 8 | 128d | nn.Embedding | Phase 2 |
| Semantic Suffixes | 15 | 128d | nn.Embedding | Phase 2 |
| Grammatical Prefixes | 4 | 16d | register_buffer | Never |
| Aspectual Suffixes | 13 | 16d | register_buffer | Never |
| Endings | 17 | 16d | register_buffer | Never |

**Comment by marctjones** [comment]
Superseded by #65 (Epic: Fundamento-Centered Training). Architecture redesigned around Fundamento-first initialization.
